model/storage/NewSqlStorage.php

```diff
@@ -0,0 +1,88 @@
+<?php
+
+/**
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU General Public License
+ * as published by the Free Software Foundation; under version 2
+ * of the License (non-upgradable).
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program; if not, write to the Free Software
+ * Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ *
+ * Copyright (c) 2015 (original work) Open Assessment Technologies SA;
+ *
+ */
+
+namespace oat\taoRevision\model\storage;
+
+use Exception;
+use oat\taoRevision\model\Revision;
+use Ramsey\Uuid\Uuid;
+
+class NewSqlStorage extends RdsStorage
+{
+
+    const DATA_RESOURCE_ID = 'id';
+
+    /**
+     * @param string $resourceId
+     * @param string $version
+     * @param string $created
+     * @param string $author
+     * @param string $message
+     * @param Triple[] $data
+     * @return Revision
+     * @throws Exception
+     */
+    public function addRevision($resourceId, $version, $created, $author, $message, $data)
+    {
+        $this->getPersistence()->insert(
+            self::REVISION_TABLE_NAME,
+            [
+                self::REVISION_RESOURCE => (string)$resourceId,
+                self::REVISION_VERSION => (string)$version,
+                self::REVISION_USER => (string)$author,
+                self::REVISION_MESSAGE => (string)$message,
+                self::REVISION_CREATED => (string)$created
+            ]
+        );
+
+        $revision = new Revision($resourceId, $version, $created, $author, $message);
+
+        if (!empty($data)) {
+            $this->saveData($revision, $data);
+        }
+        return $revision;
+    }
+
+    /**
+     * @param Revision $revision
+     * @param array $data
+     * @return bool
+     * @throws Exception
+     */
+    protected function saveData(Revision $revision, $data)
+    {
+        $dataToSave = [];
+
+        foreach ($data as $triple) {
+            $dataToSave[] = [
+                self::DATA_RESOURCE_ID => (string)Uuid::uuid4(),
```

Comment:
Can you reuse existing wrapper for this lib? 
like this https://github.com/oat-sa/generis/blob/c063f936ae5d56809e3efd07d68c57cc81c5267a/helpers/UuidPrimaryKeyTrait.php#L28 
That is not yet merged to develop, but we should consider that possiblity

Reply:
for sure this will be done inside the BackOffice package 
but not inside this PR